digiroad2-viite/src/main/scala/fi/liikennevirasto/viite/process/RoadAddressChangeInfoMapper.scala

```diff
@@ -145,19 +145,34 @@ object RoadAddressChangeInfoMapper extends RoadAddressMapper {
   def resolveChangesToMap(roadAddresses: Map[(Long, Long), LinkRoadAddressHistory], changedRoadLinks: Seq[RoadLink],
                           changes: Seq[ChangeInfo]): Map[Long, LinkRoadAddressHistory] = {
     val current = roadAddresses.flatMap(_._2.currentSegments).toSeq
-    val sections = partition(current)
-    val originalAddressSections = groupByRoadSection(sections, roadAddresses.values)
+    val history = roadAddresses.flatMap(_._2.historySegments).toSeq
+    val currentSections = partition(current)
+    val historySections = partition(history)
+    val originalAddressSections = groupByCurrentRoadSection(currentSections, roadAddresses.values)
+    val historyAddressSections = groupByHistoryRoadSection(historySections, roadAddresses.values)
     preTransferCheckBySection(originalAddressSections)
     val groupedChanges = changes.groupBy(_.vvhTimeStamp).values.toSeq
     val appliedChanges = applyChanges(groupedChanges.sortBy(_.head.vvhTimeStamp), roadAddresses.mapValues(_.allSegments))
-    val result = postTransferCheckBySection(groupByRoadSection(sections, appliedChanges.values.map(
-      s => LinkRoadAddressHistory(s.partition(_.endDate.isEmpty)))), originalAddressSections)
+    val mappedChanges = appliedChanges.values.map(
+      s => LinkRoadAddressHistory(s.partition(_.endDate.isEmpty)))
+    val currentSectionChanges = groupByCurrentRoadSection(currentSections, mappedChanges)
+    val historySectionChanges = groupByHistoryRoadSection(historySections, mappedChanges)
+    val result = postTransferCheckBySection(currentSectionChanges, historySectionChanges, originalAddressSections, historyAddressSections)

```

Comment:
Instead of having all of that duplicated code, shouldn't we have a separated method that then can be called to current sections and then by history sections?

Reply:
They are actually different methods. But i will still create one method to call them both.

Reply:
I will also create some tests for it